src/speedruncompy/endpoints.py

```diff
@@ -19,19 +19,53 @@ class GetGameLeaderboard2(GetRequest[r_GetGameLeaderboard2], BasePaginatedReques
     - @categoryId
 
     ### Optional:
-    - @levelId: If `categoryId` refers to a level category. # TODO: check if mandatory
-    - @video: `VideoFilter` = 1 (=Required (!))
-    - @timer: TimerName to sort by
-    - @obsolete: `ObsoleteFilter` = 0 # TODO: check
+    - @dateFrom: datestr = Release date # Needs to be in "YYYY-MM-DD" format
+    - @dateTo: datestr = Now # Needs to be in "YYYY-MM-DD" format
+    - @emulator: `EmulatorFilter`
+    - @levelId: If `categoryId` refers to a level category.
+    - @obsolete: `ObsoleteFilter` = 0
     - @platformIds
     - @regionIds
-    - @dateFrom: datestr
-    - @dateTo: datestr
+    - @timer: TimerName to sort by
+    - @verified: `VerifiedFilter` = 1 # If runs other than verified should be included
+    - @values: A list of `VarValues`
+    - @video: `VideoFilter` = 1 (=Required (!))
     - @page
     """
-    def __init__(self, gameId: str, categoryId: str, _api: SpeedrunClient | None = None, **params) -> None:
+    def __init__(
```

Comment:
I would rather leave out optional keywords in favour of (reformatted to ReST for broader editor support, damn you VSCode) docstrings, since all keyword arguments are passed to the API, which feels like a smell (if a default on SRC changes, the library will not notice it).

However since the work appears to be mostly done already I could be convinced otherwise.

Reply:
Their default type is None, so, I would have thought that these wouldn't be sent to the API.
Also, a doc string's autocomplete won't be as good as these types. And you have these optional keywords in other parts of this code.